lib/improver/utilities/cube_manipulation.py

```diff
@@ -289,30 +300,40 @@ def _equalise_cube_attributes(cubes):
     # without raising a warning message
     silent_attributes = ['history', 'title', 'mosg__grid_version']
     unmatching_attributes = compare_attributes(cubes)
+
     if len(unmatching_attributes) > 0:
         for i, cube in enumerate(cubes):
             # Remove ignored attributes.
             for attr in silent_attributes:
                 if attr in unmatching_attributes[i]:
                     cube.attributes.pop(attr)
                     unmatching_attributes[i].pop(attr)
-            # Add associated model_id if model_configurations do not match.
-            if "mosg__model_configuration" in unmatching_attributes[i]:
-                model_title = cube.attributes.pop('mosg__model_configuration')
-                for key, val in MODEL_ID_DICT.items():
-                    if val == model_title:
-                        break
-                new_model_id_coord = build_coordinate([key],
+
+            # If a model_id_attr has been specified, assume we are trying to
+            # grid blend and throw an error at this stage if the model ID
+            # does not match that on the cubes to be blended.
```

Comment:
`... throw an error if model_id_attr is not present on all cubes to be blended`

Reply:
Updated.